# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/animation/AnimationEventDispatcher.h
# Commit: 6f62586a139b
# Full Hash: 6f62586a139b1fcc4817dddd924400958d89a6bb
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 2
# Description:
#   Bug 1846516 - [css-properties-values-api] Animate custom properties in a discrete way. r=firefox-style-system-reviewers,emilio
#   
#   This patch also updates the bug ID for a FIXME leftover from bug 1840478
#   to bug 1869476, since the same FIXME is added in D190758.
#   
# ==============================================================================

diff -r 729abedf9bda -r 6f62586a139b dom/animation/AnimationEventDispatcher.h
--- a/dom/animation/AnimationEventDispatcher.h	Mon Dec 18 09:24:08 2023 +0000
+++ b/dom/animation/AnimationEventDispatcher.h	Mon Dec 18 09:24:08 2023 +0000
@@ -46,10 +46,12 @@
         oncompositor.AppendLiteral(", ");
       }
       properties.Append(nsCSSProps::GetStringValue(property));
-      oncompositor.Append(nsCSSProps::PropHasFlags(
-                              property, CSSPropFlags::CanAnimateOnCompositor)
-                              ? "true"
-                              : "false");
+      oncompositor.Append(
+          property != eCSSPropertyExtra_variable &&
+                  nsCSSProps::PropHasFlags(property,
+                                           CSSPropFlags::CanAnimateOnCompositor)
+              ? "true"
+              : "false");
     }
 
     aWriter.StringProperty("properties", properties);
@@ -81,9 +83,11 @@
                                    nsCSSPropertyID aProperty, bool aCanceled) {
     aWriter.StringProperty("Target", aTarget);
     aWriter.StringProperty("property", nsCSSProps::GetStringValue(aProperty));
-    aWriter.BoolProperty("oncompositor",
-                         nsCSSProps::PropHasFlags(
-                             aProperty, CSSPropFlags::CanAnimateOnCompositor));
+    aWriter.BoolProperty(
+        "oncompositor",
+        aProperty != eCSSPropertyExtra_variable &&
+            nsCSSProps::PropHasFlags(aProperty,
+                                     CSSPropFlags::CanAnimateOnCompositor));
     if (aCanceled) {
       aWriter.BoolProperty("Canceled", aCanceled);
     }